### ISSUE TITLE
Unable to find analytics.yml config file

### ISSUE THREAD

**Issue by dalareo** [OpenSaasAU/edx-docker-ansible#2]
I get this error when building:

````
fatal: [analytics]: FAILED! => {"changed": true, "cmd": "DB_MIGRATION_USER='root' DB_MIGRATION_PASS='password' /edx/app/analytics/venvs/analytics/bin/python ./manage.py migrate --noinput", "delta": "0:00:00.204642", "end": "2016-12-28 03:37:42.064722", "failed": true, "rc": 1, "start": "2016-12-28 03:37:41.860080", "stderr": "Traceback (most recent call last):\n File "./manage.py", line 10, in \n execute_from_command_line(sys.argv)\n File "/edx/app/analytics/venvs/analytics/local/lib/python2.7/site-packages/django/core/management/init.py", line 353, in execute_from_command_line\n utility.execute()\n File "/edx/app/analytics/venvs/analytics/local/lib/python2.7/site-packages/django/core/management/init.py", line 302, in execute\n settings.INSTALLED_APPS\n File "/edx/app/analytics/venvs/analytics/local/lib/python2.7/site-packages/django/conf/init.py", line 55, in getattr\n self._setup(name)\n File "/edx/app/analytics/venvs/analytics/local/lib/python2.7/site-packages/django/conf/init.py", line 43, in _setup\n self._wrapped = Settings(settings_module)\n File "/edx/app/analytics/venvs/analytics/local/lib/python2.7/site-packages/django/conf/init.py", line 99, in init\n mod = importlib.import_module(self.SETTINGS_MODULE)\n File "/usr/lib/python2.7/importlib/init.py", line 37, in import_module\n import(name)\n File "/edx/app/analytics/analytics/analyticsdataserver/settings/production.py", line 31, in \n with open(CONFIG_FILE) as f:\nIOError: [Errno 2] No such file or directory: '/edx/etc/analytics.yml'", "stdout": "", "stdout_lines": [], "warnings": []}
````

**Comment by borisno2** [comment]
Thanks @dalareo 
Here is a copy and paste from issue #1 for reference

Looks like it can't find the config file /edx/etc/analytics.yml this file should have been created during the edx_service role dependency. This file is stored in the docker image itself not on local disk. If you scroll back up in the logs can you see something like TASK [edx-service: Write out app config file]? has it changed or skipped? Can you docker cp ansible_analytics_1:/edx/etc/analytics.yml . if not can you run docker cp ansible_analytics_1:/edx/etc/analytics_api.yml .?

**Comment by dalareo** [comment]
TASK [edx-service: Write out app config file] has skipped. I have no /edx/etc folder!! just /edx/app.

**Comment by borisno2** [comment]
@dalareo I just pushed a new edxapp_vars.yml.sample, if you rename this one to override your current edxapp_vars.yml it should fix the issue.
Let me know how you go.
Thanks

**Comment by dalareo** [comment]
I am rebuilding. What is the change in edxapp_vars.yml that avoid the issue? I am trying to understand the app as much as possible...
Is there any option to start rebuilding from the error? Rebuilding takes a lot of hours in my PC...-

**Comment by borisno2** [comment]
@dalareo Yes the long rebuilding process, takes along time for me too... I haven't found a way to restart where it left off from unfortunately.

The issue was that the analytics role has the edx_service role as a dependancy which contains the task that creates /edx/etc/analytics.yml however it had a conditional to say it only run when the variable  edx_service_config is run. this variable is passed from the analytics_api role to the edx_service role in meta/main.yml and get it from the default variable of ANALYTICS_API_SERVICE_CONFIG. I had changed a this varible so that I could use TLS in commit see commit https://github.com/OpenSaasAU/edx-docker-ansible/commit/70870da1ec208bcb3966df04e37bf777ca8e8453#diff-58e4d11120eff03c7713a241659c8079 I updated my personal edxapp_vars.yml but not that sample and because ANALYTICS_API_EMAIL_USE_TLS was null which then made ANALYTICS_API_SERVICE_CONFIG null which then made edx_service_config which meant the task to create /edx/etc/analytics.yml did not run..
Hope this makes sense..
Thanks!
